modules/core/arrow-core-data/src/main/kotlin/arrow/core/Either.kt

```diff
@@ -46,7 +46,7 @@ sealed class Either<out A, out B> : EitherOf<A, B> {
    * @param ifRight the function to apply if this is a [Right]
    * @return the results of applying the function
    */
-  inline fun <C> fold(ifLeft: (A) -> C, ifRight: (B) -> C): C = when (this) {
+  fun <C> fold(ifLeft: (A) -> C, ifRight: (B) -> C): C = when (this) {
```

Comment:
I would keep it here just so people can call `suspend` because the implementation doesn't prevent it.

Reply:
We can keep it inline here and in all autofolds because the purity plugin will ensure these do not actually accept suspend but users looking for fair optimization use and not just a hack over suspend can still benefit from inlining